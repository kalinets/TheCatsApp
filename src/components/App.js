
import React from 'react'
import { BrowserRouter, Switch, Route } from 'react-router-dom'
import withAuthentication from '../hoc/withAuthentication'
import * as Routes from '../constants/routes'
import Favourites from './Favourites/Favourites'
import ViewCat from './ViewCat/ViewCat'
import NavBar from './NavBar/NavBar'
import OfflineBanner from './OfflineBanner/OfflineBanner'
import WelcomePage from './WelcomePage/WelcomePage'
import NotFound from './NotFound/NotFound'
import SignInPage from './SignInPage/SignInPage'
import SignUpPage from './SignUpPage/SignUpPage'
import 'rsuite/dist/styles/rsuite.min.css'

const App = () => (
  <BrowserRouter>
    <NavBar />
    <h1>The Cats App by @jason</h1>
    <div className="inner">
      <Switch>
        <Route exact path={Routes.HOME} component={WelcomePage} />
        <Route path={Routes.FAVOURITES} component={Favourites} />
        <Route path={Routes.SIGN_UP} component={SignUpPage} />
        <Route path={Routes.SIGN_IN} component={SignInPage} />
        <Route path={Routes.VIEW_CAT} component={ViewCat} />
        <Route path="*" component={NotFound} />
      </Switch>
    </div>
    <OfflineBanner />
  </BrowserRouter>
)

export default withAuthentication(App)
