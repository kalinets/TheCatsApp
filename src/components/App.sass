.app-title
  color: green
  text-align: center
