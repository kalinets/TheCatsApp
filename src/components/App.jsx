import React, { Component } from 'react'
import ShowCat from './ShowCat/ShowCat'
import './App.sass'

class App extends Component {
  state = {
    isOnline: true,
  }

  componentDidMount() {
    window.addEventListener('offline', this.connectionHandler)
    window.addEventListener('online', this.connectionHandler)
  }

  connectionHandler = () => this.setState({ isOnline: navigator.onLine ? true : false })

  render() {
    const { isOnline } = this.state
    return (
      <>
        <h1 className="app-title">The Cats App by @jason</h1>
        <ShowCat isOnline={isOnline} />
        {!isOnline && <div className="app-offline"><strong>You are offline</strong></div>}
      </>
    )
  }
}

export default App
