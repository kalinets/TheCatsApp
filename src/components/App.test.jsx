import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

configure({ adapter: new Adapter() })
describe('<App />', () => {
  test('renders without crashing', () => {
    shallow(<App />)
  })
  test('matches snapshot', () => {
    expect(shallow(<App />)).toMatchSnapshot()
  })
})
