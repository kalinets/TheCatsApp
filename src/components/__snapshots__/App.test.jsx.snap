// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<ContextProvider
  value={null}
>
  <App />
</ContextProvider>
`;
